gempython/tools/amc_user_functions_xhal.py

```diff
@@ -0,0 +1,343 @@
+import os, signal, sys
+sys.path.append('${GEM_PYTHON_PATH}')
+
+from ctypes import *
+from gempython.utils.gemlogger import colormsg
+from gempython.utils.nesteddict import nesteddict
+from gempython.utils.wrappers import envCheck
+from time import sleep
+
+from rw_reg import Node, parseInt, parseXML
+
+import logging
+
+gMAX_RETRIES = 5
+gRetries = 5
+
+class colors:
```

Comment:
this is defined in `utils/gemlogging.py` or somewhere in there

Reply:
Okay, I'll remove it and change to an import.

Reply:
this will be also in reg_utils... or we might need some moar refactoring

Reply:
regarding this PR, this is probably the only comment that *must* be implemented, as the others will be future cleanup